gallery_dl/extractor/tiktok.py

```diff
@@ -9,10 +9,9 @@
 from .common import Extractor, Message
 from .. import exception, text, util, ytdl
 from re import compile, escape, IGNORECASE
-from datetime import datetime
 
 BASE_PATTERN = r"(?:https?://)?(?:www\.)?tiktok(?:v?)\.com"
-USER_PATTERN = BASE_PATTERN + r"/+@([\w.]{0,23}\w)?"
+USER_PATTERN = BASE_PATTERN + r"/+@([\w\-_\.]+)?"
```

Comment:
Could you provide further context behind this change? Typically usernames can't have the `-` character so I'm wondering under what context this is allowed? Your commit mentions `update USER_PATTERN to allow id usernames`, what are "ID usernames?" It would be good to add at least one or two test cases for this type of username.

TikTok usernames can't end with the `.` character, and they can only be 24 characters in length, which is why `[\w.]{0,23}` and `\w` are separate here.

Reply:
Thank you for replying. The id field in the username is actually a channel_id, which you can find in the info.json file generated by yt-dlp. For example, a TikTok account ID might look like:
MS4wLjABAAAAv7iSuuXDJGDvJkmH_vz1qkDZYo1apxgzaxdBSeIuPiM
Some user IDs are even longer.

I use the channel_id instead of the username because usernames can change over time.

Regarding the . issue, some usernames start with a dot, so that was my mistake in the regex.

Reply:
Thanks for the explanation! I'll have a play around with channel IDs, then I can update the REGEX and any other pieces of code as necessary tomorrow at the latest (assuming you don't get round to updating the PR by then).

Reply:
```suggestion
USER_PATTERN = BASE_PATTERN + r"/@([\w_.-]+)"
```
* No multiple /// (`/+`)
* No empty username (`(...)?`)
* Backslash escapes in `[...]` are not necessary
  (except `-`, but not when it is the last character)

We could also just use a generic `[^/?#]+` here that matches basically everything.